includes/core-changes/corefx/5.0/lastindexof-improved-handing-of-empty-values.md

```diff
@@ -0,0 +1,59 @@
+### LastIndexOf has improved handling of empty search strings
+
+<xref:System.String.LastIndexOf%2A?displayProperty=nameWithType> and related APIs now return correct values when searching for a zero-length (or zero-length equivalent) substring within a larger string.
+
+#### Change description
+
+In .NET Framework and .NET Core 1.0 - 3.1, <xref:System.String.LastIndexOf%2A?displayProperty=nameWithType> and related APIs might return an incorrect value when the caller searches for a zero-length substring.
+
+```csharp
+Console.WriteLine("Hello".LastIndexOf("")); // prints '4' (incorrect)
+
+ReadOnlySpan<char> span = "Hello";
+Console.WriteLine(span.LastIndexOf("")); // prints '0' (incorrect)
+```
+
+Starting with .NET 5.0, these APIs return the correct value for `LastIndexOf`.
+
+```csharp
+Console.WriteLine("Hello".LastIndexOf("")); // prints '5' (correct)
+
+ReadOnlySpan<char> span = "Hello";
+Console.WriteLine(span.LastIndexOf("")); // prints '5' (correct)
+```
+
+In these examples, `5` is the correct answer because `"Hello".Substring(5)` and `"Hello".AsSpan().Slice(5)` both produce an empty string, which is trivially equal to the empty substring that is sought.
+
+#### Reason for change
+
+This change was part of an overall bug fixing effort around string handling for .NET 5. It also helps unify our behavior between Windows and non-Windows platforms. For more information, see [dotnet/runtime#13383](https://github.com/dotnet/runtime/issues/13383) and [dotnet/runtime##13382](https://github.com/dotnet/runtime/issues/13382).
+
+#### Version introduced
+
+5.0
+
+#### Recommended action
+
+You don't need to take any action. The .NET 5.0 runtime provides the new behaviors automatically.
+
+There is no compatibility switch to restore the old behavior.
+
+#### Category
+
+Core .NET libraries
+
+#### Affected APIs
+
+- <xref:System.String.LastIndexOf%2A?displayProperty=fullName>
```

Comment:
This means all overloads? but I don't think it's correct. For example, the overload that takes a `char` isn't affected?

Reply:
Depends on the overload. I don't think it's worth discussing this kind of nuance in the doc, though.

Reply:
@GrabYourPitchforks I think at least the commented section needs to mention the specific overloads, as it's used by API compat tool.

Reply:
I'm going to leave this as-is. The API compat tool will at least not miss any occurrences by specifying all overloads. If you know exactly which overloads are affected and are up to it, feel free to create a new PR.